### PR TITLE
Bump Android tooling

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
Bump Android tooling

## :bulb: Motivation and Context
Closes https://github.com/getsentry/sentry-react-native/issues/1515
Closes https://github.com/getsentry/sentry-react-native/issues/1512

## :green_heart: How did you test it?
e2e test

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -45,7 +45,8 @@ jobs:
       - name: Run Android Emulator
         uses: reactivecircus/android-emulator-runner@v2
         with:
-          api-level: 29
+          api-level: 30
+          ndk: 21.4.7075529
```
Comment (marandaneto) [inline .github/workflows/e2e.yml:49]:
```suggestion
          api-level: 29
```

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v1
         with:
-          node-version: '12'
+          node-version: "12"
```
Comment (marandaneto) [inline .github/workflows/e2e.yml:23]: VS Code did it automatically

File: CHANGELOG.md [hunk 3]
```diff
@@ -114,7 +118,7 @@
 
 ## 2.2.1
 
-- build(js): Bump @sentry/* dependencies on javascript to 6.2.0 #1354
+- build(js): Bump @sentry/\* dependencies on javascript to 6.2.0 #1354
```
Comment (marandaneto) [inline CHANGELOG.md:121]: VS Code did it automatically

File: android/build.gradle [hunk 4]
```diff
@@ -5,23 +5,17 @@ def safeExtGet(prop, fallback) {
...
         versionName "1.0"
         ndk {
             abiFilters "x86", "armeabi-v7a", "x86_64", "arm64-v8a"
         }
     }
-
-    compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
-    }
```
Comment (marandaneto) [inline android/build.gradle:24]: AGP >= 4.2.x is the default value

**Review by HazAT — APPROVED** [review]

File: sample/android/build.gradle [hunk 5]
```diff
@@ -2,17 +2,16 @@
...
+        compileSdkVersion = 30
+        targetSdkVersion = 30
     }
     repositories {
         google()
         jcenter()
     }
     dependencies {
-        classpath("com.android.tools.build:gradle:3.5.4")
+        classpath("com.android.tools.build:gradle:4.2.2")
```
Comment (marandaneto) [inline sample/android/build.gradle:14]: RN also uses 4.2.2 https://github.com/facebook/react-native/blob/main/build.gradle.kts#L16

File: android/build.gradle [hunk 6]
```diff
@@ -5,23 +5,17 @@ def safeExtGet(prop, fallback) {
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion safeExtGet('compileSdkVersion', 28)
-    buildToolsVersion safeExtGet('buildToolsVersion', '28.0.3')
+    compileSdkVersion safeExtGet('compileSdkVersion', 30)
```
Comment (marandaneto) [inline android/build.gradle:8]: RN also uses https://github.com/facebook/react-native/blob/main/ReactAndroid/build.gradle#L449

File: android/build.gradle [hunk 7]
```diff
@@ -5,23 +5,17 @@ def safeExtGet(prop, fallback) {
...
 
     defaultConfig {
         minSdkVersion safeExtGet('minSdkVersion', 16)
-        targetSdkVersion safeExtGet('targetSdkVersion', 28)
+        targetSdkVersion safeExtGet('targetSdkVersion', 30)
```
Comment (marandaneto) [inline android/build.gradle:12]: RN uses 28 https://github.com/facebook/react-native/blob/13107fa3d0bcb41d1e91d676f2886c575c599bc2/ReactAndroid/build.gradle#L457 but their templates 30, https://github.com/facebook/react-native/blob/ae494e7ce199cc5d524f791d45ddce51535cdadb/template/android/build.gradle#L8 so its compatible

**Pull request merged**